### PR TITLE
Vendor update

### PR DESCRIPTION
**Pull request by AutomatedTester**
This updates tools/third_party/pytest and tools/third_party/py to their latest.

This adds `attrs`, `funcsigs`, and `pluggy` as dependecies which are needed in pytest in a `third_party` directory.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/8535)

### CODE REVIEW

**Comment by AutomatedTester** [comment]
I am happy to redo this if we dont want the history from subtrees.

**Comment by w3c-bots** [comment]
# Build BROKEN

Started: 2017-12-01 14:08:30
Finished: 2017-12-01 14:21:46


<h2>Failing Jobs</h2>
<ul>

<li>tools_unittest in py27</li>

<li>tools_unittest in py36</li>

<li>tools_unittest in pypy</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21348)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/310050550)

**Comment by gsnedders** [comment]
@AutomatedTester we've previously avoided history from subtrees

**Comment by jgraham** [comment]
Yeah, I think we should do this with `--squash`

**Comment by jgraham** [comment]
Also if you have to vendor in a bunch of deps, can we create `tools/thirdparty` and we can eventually figure out how to move most stuff to there.

**Comment by AutomatedTester** [comment]
ok, will update accordingly

**Comment by AutomatedTester** [comment]
@jgraham @gsnedders I've updated with squashed subtrees and have moved pytest and it's dependencies into `third_party`

File: tools/wptrunner/wptrunner/executors/pytestrunner/runner.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def record_skip(self, report):
     def record(self, test, status, message=None, stack=None):
         if stack is not None:
             stack = str(stack)
-        new_result = (test, status, message, stack)
+        new_result = (test.split("/")[-1], status, message, stack)
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/executors/pytestrunner/runner.py]: This should perhaps be `os.path.sep`. But why do we put the test name in the subtest name at all? That just makes things fragile under renames. Can we split on `::` or whatever the separator between the path and the function name is?

**Review by jgraham — CHANGES_REQUESTED** [review]

File: tools/wptrunner/wptrunner/executors/pytestrunner/runner.py (hunk 1 above)
Comment (AutomatedTester) [reply tools/wptrunner/wptrunner/executors/pytestrunner/runner.py]: Agree on the `os.path.sep`.

As for the names of this, we currently use the file name is the metadata, https://dxr.mozilla.org/mozilla-central/source/testing/web-platform/meta/webdriver/tests/sessions/get_timeouts.py.ini#2

Changing this works might mean we have to redo all the metadata. Is that something we dont mind about (Will this break other vendors WPT?)

**Comment by AutomatedTester** [comment]
Try run on Mozilla infra with these code changes https://treeherder.mozilla.org/#/jobs?repo=try&revision=66c48462a122

**Comment by jgraham** [comment]
It might break all the metadata, but only subtest names and I think it will make things better in the long run. We get semi-frequent breakage when files are moved around as it is, and we can prevent that.

**Comment by gsnedders** [comment]
You need to update `tools/tox.ini` to have the right set of exclude paths, I think, to get Travis CI passing.

**Comment by jgraham** [comment]
Seems like the `pywebsocket` changes are badness.

**Comment by AutomatedTester** [comment]
@jgraham I havent updated the `pywebsocket` dependencies as it doesnt look like Google has pushed anything to that repo in years.

**Comment by AutomatedTester** [comment]
@gsnedders updated the files used for testing. We need docs on how to run tests properly for `tools`

**Comment by AutomatedTester** [comment]
@jgraham the only thing that I can find from my try run is https://github.com/w3c/web-platform-tests/commit/d8ca91efb791173048ea51592625915e690c7a2e which looks quite innocuous so guidance here would be appreciated

**Comment by AutomatedTester** [comment]
Latest [try run](https://treeherder.mozilla.org/#/jobs?repo=try&revision=cc5829c3596c) with this PR and cherry pick of https://github.com/w3c/web-platform-tests/commit/775e4b338a908fac840d118c0350d53f9da558f9

**Comment by AutomatedTester** [comment]
current [try](https://treeherder.mozilla.org/#/jobs?repo=try&revision=5687db1144b5) looks good for a merge

**Comment by AutomatedTester** [comment]
current [try](https://treeherder.mozilla.org/#/jobs?repo=try&revision=cbeb2fc81406) looks good for a merge

**Review by jgraham — APPROVED** [review]

**Pull request merged**